src/pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -55,35 +54,28 @@ def dispatch(self, trainer: "pl.Trainer") -> None:
             self._connected = True
         return super().dispatch(trainer)
 
-    def backward(
+    def backward(  # type: ignore[override]
         self,
+        tensor: Tensor,
         model: "pl.LightningModule",
-        closure_loss: Tensor,
-        optimizer: Optional[Optimizer],
-        optimizer_idx: Optional[int],
         *args: Any,
         **kwargs: Any,
     ) -> None:
         """Run before precision plugin executes backward.
 
         Args:
+            tensor: the loss value obtained from the closure
             model: the model to be optimized
-            closure_loss: the loss value obtained from the closure
-            optimizer: current optimizer being used. ``None`` if using manual optimization
-            optimizer_idx: the index of the current optimizer. ``None`` if using manual optimization
         """
+        optimizer, optimizer_idx, *args = args
```

Comment:
can't the optimizer and index be passed as kwargs? Does the lite class take care of this so that they are always passed as args here?

To be safe we would need to do something like

```python
if "optimzer" in kwargs:
    optimizer = kwargs.pop("optimizer")
else:
   optimizer, *args = args

if "optimizer_idx" in kwargs:
    optimizer_idx = kwargs.pop("optimizer_idx")
else:
    optimizer_idx, *args = args
```

Reply:
The strategy calls the precision plugin like this:

        self.precision_plugin.backward(closure_loss, self.lightning_module, optimizer, optimizer_idx, *args, **kwargs)

This is why they are positional. There is a contract there that this is the way to  pass them in. I wanted to minimize breaking change.

Reply:
> Does the lite class take care of this so that they are always passed as args here?

The lite class doesn't have a definition for optimizer or idx.